apps/sms/views/shared/test/unit/utils_test.js

```diff
@@ -1153,6 +1153,131 @@ suite('Utils', function() {
     });
   });
 
+  suite('Utils.throttle', function() {
+    test('right typeof parameters', function() {
+      assert.throw(() => Utils.throttle(null), 
+                   Error, 
+                   'func must be a Function');
+      assert.throw(() => Utils.throttle(() => {}, null),
+                   Error, 
+                   'delay must be a positive number');
+      assert.throw(() => Utils.throttle(() => {}, -1),
+                  Error, 
+                  'delay must be a positive number');
+    });
+
+    test('parameters are passed to the throttled func', function() {
+      var dummyObjectA = {};
+      var dummyObjectB = {};
+
+      var func = function() {
+        assert.equal(arguments[0], dummyObjectA);
+        assert.equal(arguments[1], dummyObjectB);
+      };
+
+      var throttlerFunc = Utils.throttle(func, 100);
+      
+      throttlerFunc(dummyObjectA, dummyObjectB);
+    });
+
+    test('testing this in the throttled func', function() {
+      var self = this;
+      var dummyObject = {};
+
+      var throttlerFunc = Utils.throttle(function() {
+        assert.equal(this, dummyObject);
+      }, 100);
+      throttlerFunc.call(dummyObject);
+
+      var throttlerFunc2 = Utils.throttle(()=>{
+        assert.equal(this, self);
+      }, 100);
+      throttlerFunc2.call(dummyObject);
+    });
+
+    test('preventing first call', function() {
+      var delay = 100;
+      var firstCallFuncToExecute = sinon.stub();
+      var firstCallThrottleFuncToExecute = 
+        Utils.throttle(firstCallFuncToExecute, delay);
+      
+      var notFirstCallFuncToExecute = sinon.stub();
+      var notFirstCallThrottleFuncToExecute = 
+      Utils.throttle(notFirstCallFuncToExecute, delay, true);
+
+      firstCallThrottleFuncToExecute();
+      sinon.assert.calledOnce(firstCallFuncToExecute);
+
+      notFirstCallThrottleFuncToExecute();
+      sinon.assert.notCalled(notFirstCallFuncToExecute);
+    });
+
+    test('preventing last call', function() {
+      var clock = sinon.useFakeTimers(Date.now());
```

Comment:
Same as https://github.com/mozilla-b2g/gaia/pull/31677/files#r38888591 you can setup the useFakeTimers at the top of the Utils.throttle suite and use this.sinon.clock.tick in the test. You can reference the debounce unit test part : https://github.com/mozilla-b2g/gaia/pull/31677/files#diff-52ec1815f0d22fe9a17ef2259891a61dR1127